File: PyProject/leetcode/maximum-depth-of-binary-tree.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class MaximumDepthOfBinaryTree:
    def maxDepth(self, root: TreeNode) -> int:
        return self.helper(root, 0)

    def helper(self, root: TreeNode, level: int) -> int:
        if not root:
            return level
        left_level = self.helper(root.left, level + 1)
        right_level = self.helper(root.right, level + 1)
        if left_level > right_level:
            return left_level
        else:
            return right_level

File: GoProject/leetcode/dfs/maximum-depth-of-binary-tree.go
package dfs

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	right := maxDepth(root.Right) + 1
	left := maxDepth(root.Left) + 1
	if right > left {
		return right
	}
	return left

}
